knotx-server/src/main/java/io/knotx/server/KnotxGatewayContextHandler.java

```diff
@@ -61,6 +66,12 @@ private void handleRoute(final RoutingContext context, final String address) {
     if (StringUtils.isNotBlank(bodyAsString)) {
       knotContext.setFragments(Collections.singletonList(Fragment.raw(bodyAsString)));
     }
+    Set<FileUpload> fileUploads = context.fileUploads();
+    if (fileUploads != null) {
+      knotContext.setFilesData(fileUploads.stream().map(fileUpload -> {
```

Comment:
You may do it also like this:
`knotContext.setFilesData(fileUploads.stream().map(FileData::new).collect(Collectors.toSet()))`